synapse/util/check_dependencies.py

```diff
@@ -128,6 +128,21 @@ def _incorrect_version(
         )
 
 
+def _no_reported_version(
+    requirement: Requirement, extra: Optional[str] = None
+):
+    if extra:
+        return (
+            f"Synapse {VERSION} needs {requirement} for {extra}, "
+            f"but can't determine {requirement.name}'s version."
+        )
+    else:
+        return (
+            f"Synapse {VERSION} needs {requirement},"
```

Comment:
nit: the message is missing a space
```suggestion
            f"Synapse {VERSION} needs {requirement}, "
```